compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/GenerateClosuresForDefaultValues.java

```diff
@@ -377,8 +381,16 @@ public void visit(BLangObjectConstructorExpression objectConstructorExpression)
 
     @Override
     public void visit(BLangRecordTypeNode recordTypeNode) {
+        if (((BRecordType) recordTypeNode.getBType()).mutableType != null) {
+            BRecordTypeSymbol typeSymbol =
+                    (BRecordTypeSymbol) ((BRecordType) recordTypeNode.getBType()).mutableType.tsymbol;
+            ((BRecordTypeSymbol) recordTypeNode.getBType().tsymbol).defaultValues = typeSymbol.defaultValues;
+            recordTypeNode.restFieldType = rewrite(recordTypeNode.restFieldType, env);
```

Comment:
Why does only the rest field type need to be visited?

Reply:
Here, no need to generate closures for default values of fields. Those are already generated for to mutable type and we set default values from the mutable type.

Reply:
How is it different for the rest type vs other field types then? Wouldn't those of the rest type also be generated?

Reply:
Updated the logic.

Reply:
Not sure how this is updated?

Also why do we need to do this again? We wrap this instead at value creation, right?

Reply:
Yes. I have removed that logic.